javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferenceCreator.java

```diff
@@ -244,12 +253,68 @@ public FieldVisitor visitField(
   @Override
   public MethodVisitor visitMethod(
       int access, String name, String descriptor, String signature, String[] exceptions) {
+
+    // declared method references are not generated for advice classes, only for helper classes
```

Comment:
maybe as an optimization here, don't generate method references for private methods, <init> or <clinit> since they can't implement an abstract method

Reply:
That's a very good idea, I'll do that. Thanks!